src/test/java/com/google/sps/servlets/RentPostsServletTest.java

```diff
@@ -0,0 +1,160 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+@RunWith(JUnit4.class)
+public class RentPostsServletTest {
+  @Mock CollectionReference collectionMock;
+  @Mock Firestore dbMock;
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+  private RentPostsServlet rentPostsServlet;
+
+  @Before
+  public void setUp() throws Exception {    
+    MockitoAnnotations.initMocks(this);
+    rentPostsServlet = new RentPostsServlet();
+    rentPostsServlet.init();
+    FirestoreUtil.setDatabase(dbMock);
+  }
+
+  @Test
+  public void testDatabaseMock_MockExistsAndIsUsedByServlet() throws Exception {
+    assertNotNull(FirestoreUtil.getDatabase());
+    assertEquals(dbMock, FirestoreUtil.getDatabase());
+  }
+
+  @Test
+  public void testPost_postsSingleRentPost() throws Exception {
+    String description = "Test description";
+    String endDate = "2020-07-10";
+    String leaseType = "YEAR_LONG";
+    String numRooms = "1";
+    String price = "100";
+    String roomType = "SINGLE";
+    String startDate = "2020-07-10";
+    String title = "Test title";
+    setupDatabaseMocks();
+    setRequestParameters(description, endDate, leaseType, numRooms,
+      price, roomType, startDate, title);
+    Map<String, Object> expectedData = getExpectedRentData(description, endDate,
+      leaseType, numRooms, price, roomType, startDate, title);
+
+    rentPostsServlet.doPost(request, response);
+
+    verify(dbMock, Mockito.times(1)).collection(RENT_COLLECTION_NAME);
+    verify(collectionMock, Mockito.times(1)).add(expectedData);
+  }
+
+  @Test
+  public void testPost_requestHasUnparsableDates_postsWithNullDates() throws Exception {
+    String description = "Test description";
+    String endDate = "202020/20/10";
+    String leaseType = "YEAR_LONG";
+    String numRooms = "1";
+    String price = "100";
+    String roomType = "SINGLE";
+    String startDate = "07/10/2020";
+    String title = "Test title";
+    setupDatabaseMocks();
+    setRequestParameters(description, endDate, leaseType, numRooms,
+      price, roomType, startDate, title);
+    Map<String, Object> expectedData = 
+      getExpectedRentData(description, endDate, leaseType, numRooms, price, 
+        roomType, startDate, title);
+    expectedData.put(REQUEST_START_DATE, null);
+    expectedData.put(REQUEST_END_DATE, null);
+    
+    rentPostsServlet.doPost(request, response);
+
+    verify(dbMock, Mockito.times(1)).collection(RENT_COLLECTION_NAME);
+    verify(collectionMock, Mockito.times(1)).add(expectedData);
+  }
+  
+  private void setupDatabaseMocks() {
+    when(dbMock.collection(RENT_COLLECTION_NAME)).thenReturn(collectionMock);
```

Comment:
You can set this up in setup() function instead of creating private function and calling it everytime.